physical/postgresql/postgresql.go

```diff
@@ -213,3 +282,198 @@ func (m *PostgreSQLBackend) List(ctx context.Context, prefix string) ([]string,
 
 	return keys, nil
 }
+
+// LockWith is used for mutual exclusion based on the given key.
+func (p *PostgreSQLBackend) LockWith(key, value string) (physical.Lock, error) {
+	identity, err := uuid.GenerateUUID()
+	if err != nil {
+		return nil, err
+	}
+	return &PostgreSQLLock{
+		backend:  p,
+		key:      key,
+		value:    value,
+		identity: identity,
+	}, nil
+}
+
+func (p *PostgreSQLBackend) HAEnabled() bool {
+	return p.haEnabled
+}
+
+// Lock tries to acquire the lock by repeatedly trying to create
+// a record in the PostgreSQL table. It will block until either the
+// stop channel is closed or the lock could be acquired successfully.
+// The returned channel will be closed once the lock is deleted or
+// changed in the PostgreSQL table.
+func (l *PostgreSQLLock) Lock(stopCh <-chan struct{}) (doneCh <-chan struct{}, retErr error) {
+	l.lock.Lock()
+	defer l.lock.Unlock()
+	if l.held {
+		return nil, fmt.Errorf("lock already held")
+	}
+
+	done := make(chan struct{})
```

Comment:
`done` seems to serve no purpose in this func.

Reply:
To be pefectly honest I put my trust in the DynamoDB implemtation. My Aim was primarily to change the db specifics but keep other logic . But I agree with your analysis why create our own done channel.